plc4j/tools/connection-cache/src/main/java/org/apache/plc4x/java/utils/cache/CachedPlcConnectionManager.java

```diff
@@ -81,14 +93,16 @@ public PlcConnection getConnection(String url) throws PlcConnectionException {
         try {
             return leaseFuture.get(this.maxWaitTime.toMillis(), TimeUnit.MILLISECONDS);
         } catch (ExecutionException | InterruptedException | TimeoutException e) {
+            connectionContainer.close();
+            connectionContainers.remove(url);
             throw new PlcConnectionException("Error acquiring lease for connection", e);
         }
     }
 
-    public PlcConnection getConnection(String url, PlcAuthentication authentication) throws PlcConnectionException {
-        throw new PlcConnectionException("the cached driver manager currently doesn't support authentication");
```

Comment:
I intentionally left away the autherntication as in general we couldn't have two connections to the same plc with different authentication ... if we do this, we should probably also add the authentication information to the key of the map.

Reply:
Just leave it for the future to authorize

Reply:
Well I'd rather simply do it right, right away ... shouldn't be too much additional work ... sort of like a "url + authentication.toString()"